relationship.go

```diff
@@ -0,0 +1,53 @@
+package jsh
+
+import (
+	"fmt"
+
+	"encoding/json"
+)
+
+// Relationship represents a reference from the resource object in which it's
+// defined to other resource objects.
+type Relationship struct {
+	Links *Links                 `json:"links,omitempty"`
+	Data  ResourceLinkage        `json:"data,omitempty"`
+	Meta  map[string]interface{} `json:"meta,omitempty"`
+}
+
+// ResourceLinkage is a typedef around a slice of resource identifiers. This
+// allows us to implement a custom UnmarshalJSON.
+type ResourceLinkage []*ResourceIdentifier
+
+// ResourceIdentifier identifies an individual resource.
+type ResourceIdentifier struct {
```

Comment:
After seeing this, I was originally thinking we should make this a generic identifier, but for these cases an ID is required where as a for a top level data object it's not.